docs/authelia-from-scratch.md

```diff
@@ -0,0 +1,448 @@
+    # Some may be not relevant for your own setup.
+    add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;
+    add_header X-Content-Type-Options nosniff;
+    add_header X-Frame-Options "SAMEORIGIN";
+    add_header X-XSS-Protection "1; mode=block";
+    add_header X-Robots-Tag "noindex, nofollow, nosnippet, noarchive";
+    add_header X-Download-Options noopen;
+    add_header X-Permitted-Cross-Domain-Policies none;
+    add_header X-Forwarded-Proto https;
+    add_header X-Forwarded-Host authelia.example.com;
+    proxy_headers_hash_max_size 512;
+    proxy_headers_hash_bucket_size 128;
+    EOF
+
+    # cat <<EOF > proxy.conf
+    client_body_buffer_size 128k;
+    
+    #Timeout if the real server is dead
+    proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;
+    
+    # Advanced Proxy Config
+    send_timeout 5m;
+    proxy_read_timeout 360;
+    proxy_send_timeout 360;
+    proxy_connect_timeout 360;
+    
+    # Basic Proxy Config
+    proxy_set_header Host $host;
+    proxy_set_header X-Real-IP $remote_addr;
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header X-Forwarded-Proto $scheme;
+    proxy_set_header X-Forwarded-Host $http_host;
+    proxy_set_header X-Forwarded-Uri $request_uri;
+    proxy_set_header X-Forwarded-Ssl on;
+    proxy_redirect  http://  $scheme://;
+    proxy_http_version 1.1;
+    proxy_set_header Connection "";
+    proxy_cache_bypass $cookie_session;
+    proxy_no_cache $cookie_session;
+    proxy_buffers 64 256k;
+    
+    # If behind reverse proxy, forwards the correct IP
+    set_real_ip_from 10.0.0.0/8;
+    set_real_ip_from 172.0.0.0/8;
+    set_real_ip_from 192.168.0.0/16;
+    set_real_ip_from fc00::/7;
+    real_ip_header X-Forwarded-For;
+    real_ip_recursive on;
+    EOF
+
+    # cat <<EOF > proxy_params
+    proxy_set_header Host $http_host;
+    proxy_set_header X-Real-IP $remote_addr;
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header X-Forwarded-Proto $scheme;
+    proxy_set_header X-Forwarded-host authelia.example.com;
+    proxy_http_version 1.1;
+    proxy_set_header Upgrade $http_upgrade;
+    proxy_set_header Connection "upgrade";
+    proxy_cache_bypass $http_upgrade;
+    EOF
+
+    # cat <<EOF > ssl.conf
+    resolver 8.8.8.8 valid=300s;
+    resolver_timeout 10s;
+    ssl_certificate /etc/nginx/ssl/xxx.example.com.fullchain;
+    ssl_certificate_key /etc/nginx/ssl/xxx.example.com.key;
+    ssl_protocols TLSv1.2;
+    ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
+    ssl_ecdh_curve secp384r1;
+    ssl_prefer_server_ciphers on;
+    ssl_stapling on;
+    ssl_stapling_verify on;
+    ssl_session_timeout 24h;
+    ssl_session_cache shared:SSL:50m;
+    ssl_session_tickets off;
+    EOF
+
+### (re)Start NGINX
+    # ln sites-available/authelia.conf sites-enabled/
+    # systemctl restart nginx
+
+### Install Docker
```

Comment:
You can add a section where the user can download the artifact from github and verify the shasum instead and mark the following sections as "optional" up to "Setup Authelia". This is easier and does not require to install additional packages.

Those steps are only required if one needs to build for a given non-built target.

Reply:
Add a mark like '(for building from source only)' in the title to clearly identify what is really optional.

Reply:
You can do the same for all optional sections.